README.md

```diff
@@ -29,11 +44,84 @@ $ pip install pytest-raises
 Usage
 -----
 
-Marking a test with the `@pytest.mark.raises()` decorator will mark that the code the test executes is expected to raise an error.  This is different from `@pytest.mark.xfail()` as it does not mean the test itself might fail, but instead that the "pass" for the test is that the code raises an error.
-
-It accepts an `exception` keyword argument, which is the class of error expected to be raised.
-
-It will allow tests which raise errors to pass.  The main usage is to assert that an error of a specific type is raise.
+Marking a test with the `@pytest.mark.raises()` or
+`@pytest.mark.setup_raises` decorator will mark that the code the test
+executes is **expected** to raise an error.  This is different from
+`@pytest.mark.xfail()` as it does not mean the test itself might fail, but
+instead that the "pass" for the test is that the code raises an error.
+
+It will allow tests which raise errors to pass.  The main usage is to assert
+that an error of a specific type is raise.
+
+If a test is marked with `@pytest.mark.raises` or
+`@pytest.mark.setup_raises` and it does **not** `raise` in the appropriate
+testing phase, the test will be failed.
+
+### Available Markers
+
+This extension provides two markers for different phases of `pytest`:
+
+- `@pytest.mark.raises`: for marking a function that should `raise` during
+  the `pytest_runtest_call` phase.
+    - This decorator can be used in place of the
+      [`with pytest.raises(...)` context manager](https://docs.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions).
+- `@pytest.mark.setup_raises`: for marking a function that should `raise`
+  during the `pytest_runtest_setup` phase.
+
+### Limitations on Markers
+
+1. Any test function decorated with `@pytest.mark.setup_raises` is assumed
+   to have an empty function body
+
+   ```python
+   @pytest.mark.setup_raises()
+   def test_something():
+       pass
+   ```
+
+   This is because `pytest_runtest_call` may still be executed depending on
+   what raised when.  So any code in the test function body may cause
+   erroneous errors (particularly if you are using fixtures, since the
+   fixture setup may be incomplete).
+
+   See the [`@pytest.mark.setup_raises` Examples](#pytestmarksetup_raises-examples)
+   for more information.
+
+2. Since the function body of anything decorated with
+   `@pytest.mark.setup_raises` is assumed to be empty, test functions that
+   are decorated with both `@pytest.mark.raises`and
+   `@pytest.mark.setup_raises` is **not** supported.
+
+   The implementation details of this limitation are further documented in
+   the `_pytest_raises_validation` function.
+
+### Available Parameters
+
+Both markers accept the following optional parameters:
+
+- `exception=<Some Exception Class>`: the exact exception **class** that is
+  expected to be raised.
+- `message='some string'`: a verbatim message that is expected to be in the
```

Comment:
Were these features always available? :astonished: I had no idea - that's so helpful to know

Reply:
`exception` has been around from the beginning, `message` is relatively new.